bundles/org.openhab.binding.adorne/src/main/java/org/openhab/binding/adorne/internal/hub/AdorneHubController.java

```diff
@@ -0,0 +1,546 @@
+    private static final String HUB_REST_SET_BRIGHTNESS = "{\"ID\":%d,\"Service\":\"SetZoneProperties\",\"ZID\":%d,\"PropertyList\":{\"PowerLevel\":%d}}\0";
+    private static final String HUB_REST_REQUEST_STATE = "{\"ID\":%d,\"Service\":\"ReportZoneProperties\",\"ZID\":%d}\0";
+    private static final String HUB_REST_REQUEST_ZONES = "{\"ID\":%d,\"Service\":\"ListZones\"}\0";
+    private static final String HUB_REST_REQUEST_MACADDRESS = "{\"ID\":%d,\"Service\":\"SystemInfo\"}\0";
+    private static final String HUB_TOKEN_SERVICE = "Service";
+    private static final String HUB_TOKEN_ZID = "ZID";
+    private static final String HUB_TOKEN_PROPERTY_LIST = "PropertyList";
+    private static final String HUB_TOKEN_DEVICE_TYPE = "DeviceType";
+    private static final String HUB_TOKEN_SWITCH = "Switch";
+    private static final String HUB_TOKEN_DIMMER = "Dimmer";
+    private static final String HUB_TOKEN_NAME = "Name";
+    private static final String HUB_TOKEN_POWER = "Power";
+    private static final String HUB_TOKEN_POWER_LEVEL = "PowerLevel";
+    private static final String HUB_TOKEN_MAC_ADDRESS = "MACAddress";
+    private static final String HUB_TOKEN_ZONE_LIST = "ZoneList";
+    private static final String HUB_SERVICE_REPORT_ZONE_PROPERTIES = "ReportZoneProperties";
+    private static final String HUB_SERVICE_ZONE_PROPERTIES_CHANGED = "ZonePropertiesChanged";
+    private static final String HUB_SERVICE_LIST_ZONE = "ListZones";
+    private static final String HUB_SERVICE_SYSTEM_INFO = "SystemInfo";
+
+    private @Nullable Future<?> hubController;
+    private String hubHost;
+    private int hubPort;
+    private @Nullable Socket hubSocket;
+    private @Nullable PrintStream hubOut;
+    private @Nullable Scanner hubIn;
+    private CompletableFuture<@Nullable Void> hubControllerConnected;
+    private int hubReconnectSleep; // Sleep time before we attempt re-connect
+    private ScheduledExecutorService scheduler;
+
+    private AtomicBoolean stop; // Stop the controller as soon as possible
+    private AtomicBoolean stopWhenCommandsServed; // Stop the controller once all pending commands have been served
+    private long stopTimestamp = 0; // Stop the controller after this timestamp
+
+    // When we submit commmands to the hub we don't correlate commands and responses. We simply use the first available
+    // response that answers our question. For that we maintain lists of all pending commands.
+    private List<AdorneHubController.ZoneIDFutureRecord> stateCommands;
+    private List<CompletableFuture<List<Integer>>> zoneCommands;
+    private List<CompletableFuture<String>> macAddressCommands;
+    private int commandId; // We assign increasing command ids to all rest commands to the hub for easier
+                           // troubleshooting
+
+    private @Nullable AdorneHubChangeNotify changeListener;
+    private Gson gson;
+
+    private static class ZoneIDFutureRecord {
```

Comment:
Why is this static?

Reply:
Since ZoneIDFutureRecord is just a DTO to pass state through the futures it doesn't need to be an inner class that I then have to instantiate from an outer instance. Just keeps the usage a bit lighter weight.

Reply:
Correct.